projects/graph/src/graph_demo.py

```diff
@@ -37,23 +39,31 @@ def main(numVertices=None, numEdges=None,color=None,style='square',arrows=None):
         if ranNum2 not in graphConstruct.vertices[ranNum1] and ranNum1 != ranNum2:
             graphConstruct.add_edge(ranNum1, ranNum2)
             numEdges -= 1
+    if groupColor is True:
+        for key in graphConstruct.vertices:
+            toAdd = graphConstruct.returnAllPath(key)
+            if toAdd not in colorGroups:
+                colorGroups.append(toAdd)
     graphDisplay = BokehGraph(graphConstruct.vertices)
-    graphDisplay.show(style,color,arrows)
+    graphDisplay.show(style, color, arrows, colorGroups)
 
 
 if __name__ == '__main__':
     # TODO - parse argv
     # print(argv[1])
     parser = argparse.ArgumentParser(description='Generate a graph!')
-    parser.add_argument('--vertices', nargs='?', type=int,
+    parser.add_argument('-v', '--vertices', nargs='?', type=int,
                         help='number of vertices wanted (default is random)')
-    parser.add_argument('--edges', nargs='?', type=int,
+    parser.add_argument('-e', '--edges', nargs='?', type=int,
                         help='number of edges wanted (default is random)')
-    parser.add_argument('--color', nargs='?', type=str,
+    parser.add_argument('-c', '--color', nargs='?', type=str,
                         help='pick the output color (specify color, or use viridis for random colors or use connectedOnly for connected highlighting')
-    parser.add_argument('--style', nargs='?', type=str,
+    parser.add_argument('-s', '--style', nargs='?', type=str,
                         help='either circle,square, or random')
-    parser.add_argument('--arrows',
-                        help='if added will turn on arrows',action="store_true")
+    parser.add_argument('-a', '--arrows',
+                        help='if added will turn on arrows', action="store_true")
+    parser.add_argument('-g', '--groupColor',
+                        help='colors each connection group', action="store_true")
     args = parser.parse_args()
-    main(args.vertices, args.edges,args.color,args.style,args.arrows)
+    main(args.vertices, args.edges, args.color,
+         args.style, args.arrows, args.groupColor)
```

Comment:
This is terrific - I love that you can turn the arrows on and off! Nice work with polishing this so that not only do the color groups and styling work for connected components, but the command line correctly processes arguments for drawing graphs. Well done!